Integrate reset token validation on reset password landing

lukasaric opened the pull request:
This PR:
- Checks if `reset token` is valid when user lands on `reset password` page.
- Both, reset password & token validation routes, now are on `base` axios instance to avoid location replacement

lukasaric committed:
Added reset token validation :rocket:

- Set this check on reset password component pre load

lukasaric committed:
Reused passport token validation

lukasaric committed:
Changed success token validation res status

lukasaric committed:
Polished user api urls :lipstick:

lukasaric committed:
Token err msg improvements

lukasaric committed:
Removed redirect after timeout

lukasaric committed:
Added back to login btn on reset password page

lukasaric committed:
Improved back button text label :rocket:

lukasaric committed:
Add req rate limiter :rocket:

lukasaric committed:
Scoped reset pw routes within `password` prefix

lukasaric committed:
Added missing space after parentheses

lukasaric committed:
Limited requests only on token validation

lukasaric committed:
Extracted req limiter as  configurable  mw

lukasaric committed:
Disable btn if token is invalid

lukasaric committed:
Added express slow down pkg :package:

- Implement it as `reset-password` mw

lukasaric committed:
Changed folder structure :wrench:

- Isolated request mws in shared `request` folder

lukasaric committed:
Hide reset pw form if error is occured

lukasaric committed:
Minor UI tweaks

lukasaric committed:
Changed error message

lukasaric committed:
Replaced btn with more descriptive router link

lukasaric committed:
Merge branch 'develop' into feature/reset-token-validation

underscope commented on server/user/index.js:
```diff
@@ -1,18 +1,22 @@
   .post('/forgot-password', ctrl.forgotPassword)
   .post('/reset-password', authenticate('token'), ctrl.resetPassword)
-  // Protected routes:
+  .post('/reset-token-validation', authenticate('token'), (_, res) => res.sendStatus(ACCEPTED));
```
would it make sense for this to be nested under the reset-password route (i.e. `/reset-password/token-status` or similar). Add rate limiting.

underscope commented on client/store/modules/auth/actions.js:
```diff
@@ -20,6 +20,8 @@ export const resetPassword = (_, { token, password }) => {
   return api.resetPassword(token, password);
 };
 
+export const validateResetToken = (_, { token }) => api.validateResetToken(token);
```
consider moving it to `api/auth`

underscope commented on client/components/auth/ResetPassword.vue:
```diff
@@ -1,77 +1,93 @@
+          Change password
+        </v-btn>
+      </template>
+      <router-link v-else :to="{ name: 'forgot-password' }" class="forgot-password">
```
Consider rewriting this in a way you are not toggling part of the form (this should be outside).

lukasaric committed:
Addressed issues :surfer:

- Added express rate limiter :rocket:
- Attached token validation within reset-pw route
- Removed reset password management from store

lukasaric committed:
Removed forgot pw from auth actions

- Set auth/api instead directly

lukasaric committed:
Renamed reset token url method

underscope commented on client/components/auth/ResetPassword.vue:
```diff
@@ -56,22 +61,32 @@
+      return api.resetPassword(token, password)
         .then(() => this.$router.push('/'))
-        .catch(() => (this.error = 'An error has occurred!'));
+        .catch(() => (this.error = INVALID_TOKEN_ERROR));
```
Why an invalid token error at this point 🤔 ?

underscope commented on client/components/auth/ResetPassword.vue:
```diff
@@ -81,4 +96,8 @@ export default {
   background: #ececec;
 }
+
+.forgot-password .v-icon {
```
consider using icon size property

lukasaric committed:
Reset password component polish :lipstick:

lukasaric committed:
Errors enhancement :rocket:
